elkodon/src/message.rs

```diff
@@ -1,5 +1,16 @@
+use core::fmt;
+
 #[repr(C)]
 pub(crate) struct Message<Header, Data> {
     pub(crate) header: Header,
     pub(crate) data: Data,
 }
+
+impl<Header: fmt::Debug, Data: fmt::Debug> fmt::Debug for Message<Header, Data> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("Message<Header, Data>")
```

Comment:
I didn't know that this was available. I learned something :smile: 

But could it be that it has the flaw that we do not print out the type names of `Header` and `Data`?
So that this would actually provide more context information:
```rust
write!(f, "Message<{}, {}> ....", core::any::type_name::<Header>(), core::any::type_name::<Data>(), ...);
```

If this is somehow already covered by `debug_struct` I am fine, but if now, could we add this. I think this is actually helpful for logging etc.

Reply:
I have to check this. Will tell you later how the output looks like.

Reply:
This
```rs
let a = elkodon::message::Message::<bool, u64>{header: true, data: 42};
println!("{:?}", a);
println!("{:#?}", a);
```
prints
```
Message { header: true, data: 42 }
Message {
    header: true,
    data: 42,
}
```
with `#[derive(Debug)]`.

With `f.debug_struct(&format!("Message<{}, {}>", core::any::type_name::<Header>(), core::any::type_name::<Data>()))` we would get the generic parameter. But `format!` is `std` and not `core`.

When using `write(f, "...", ...)` one needs to check for the `alternate` flag and then it starts to get complicated. Not sure if it is worth the hassle.

Reply:
Then leave it as it is.